scenario/utils/relayOptimismMessage.ts

```diff
@@ -0,0 +1,93 @@
+import { DeploymentManager } from '../../plugins/deployment_manager';
+import { impersonateAddress } from '../../plugins/scenario/utils';
+import { setNextBaseFeeToZero, setNextBlockTimestamp } from './hreUtils';
+import { ethers } from 'ethers';
+import { Log } from '@ethersproject/abstract-provider';
+
+/*
+The Optimism relayer applies an offset to the message sender.
+
+applyL1ToL2Alias mimics the AddressAliasHelper.applyL1ToL2Alias fn that converts
+an L1 address to its offset, L2 equivalent.
+
+https://optimistic.etherscan.io/address/0x4200000000000000000000000000000000000007#code
+*/
+function applyL1ToL2Alias(address: string) {
+  const offset = BigInt('0x1111000000000000000000000000000000001111');
+  return `0x${(BigInt(address) + offset).toString(16)}`;
+}
+
+export default async function relayOptimismMessage(
+  governanceDeploymentManager: DeploymentManager,
+  bridgeDeploymentManager: DeploymentManager,
+  startingBlockNumber: number
+) {
+  const optimismL1CrossDomainMessenger = await governanceDeploymentManager.getContractOrThrow('baseL1CrossDomainMessenger');
```

Comment:
will the contract always be `baseL1CrossDomainMessenger`? even when we're briding to Optimism?

Reply:
Actually not, since the alias and contract address will be different for the two bridges. Good catch, I'll change all references to Optimism here (including the file name) to Base.